pat/pat.go

```diff
@@ -283,6 +284,31 @@ func (p *Pattern) String() string {
 }
 
 /*
+URLPath replaces the named matches in the Pattern with the values provided
+and returns the resulting string. It can be used to generate a URL
+that would match the given pattern with the provided variables. You must
+provide a mapping for all named matches defined in the Pattern.
+*/
+func (p *Pattern) URLPath(vars map[pattern.Variable]string) (string, error) {
```

Comment:
I might just call it `Path` (or `URL` and return a `url.URL`). Seems a bit simpler.

We could also try to make it round-trip with `Match` and return a full-blown `http.Request` but that seems like it's a bit much.